js/main.js

```diff
@@ -6,167 +6,245 @@ var TYPE_ARR = ['palace', 'flat', 'house', 'bungalo'];
 var CHECKIN_TIME_ARR = ['12:00', '13:00', '14:00'];
 var CHECKOUT_TIME_ARR = ['12:00', '13:00', '14:00'];
 var PINS_QUANTITY = 8;
+var MAIN_PIN_SIZE = {
+  width: 62,
+  height: 62,
+  tail: 22,
+};
 var PIN_SIZE = {
   width: 50,
   height: 70
 };
-var mapNode = document.querySelector('.map');
+
 var pinTemplateNode = document.querySelector('#pin').content.querySelector('.map__pin');
 var pinListNode = document.querySelector('.map__pins');
+
+var PIN_COORDINATE_LIMITS = {
+  x: pinListNode.offsetWidth,
+  maxY: 630,
+  minY: 130,
+};
+
 var fragment = document.createDocumentFragment();
 var pinsArr = [];
 
+var mapNode = document.querySelector('.map');
+var adFormNode = document.querySelector('.ad-form');
 
-mapNode.classList.remove('map--faded');
+var mapPinMainNode = document.querySelector('.map__pin--main');
+var mapFiltersNode = document.querySelector('.map__filters');
 
-// Получить рандомное число
-var getRandomNumber = function (min, max) {
-  return Math.floor(min + Math.random() * (max + 1 - min));
-};
-
-// Перемешать массив
-var shuffleArray = function (arr) {
-  var initValue;
-  var newArr = [];
-  for (var i = arr.length - 1; i >= 0; i--) {
-    var randomValue = Math.floor(Math.random() * (i + 1));
-    initValue = arr[i];
-    arr[i] = arr[randomValue];
-    arr[randomValue] = initValue;
-    newArr.push(arr[i]);
+var changeDisabledAttribute = function (collection, state) {
+  for (var i = 0; i < collection.length; i++) {
+    collection[i].disabled = state;
   }
-  return newArr;
 };
 
-// Из перемешанного массива создать массив со случайным кол-вом элементов без повторений
-var randomizeArray = function (arr) {
-  var newArr = shuffleArray(arr);
-  var shuffleArr = newArr.slice(0, getRandomNumber(1, newArr.length));
-  return shuffleArr;
-};
+changeDisabledAttribute(adFormNode.querySelectorAll('fieldset'), true);
+changeDisabledAttribute(mapFiltersNode.querySelectorAll('fieldset'), true);
+changeDisabledAttribute(mapFiltersNode.querySelectorAll('select'), true);
```

Comment:
Можно объединить в функцию. Также было бы здорово добавлять `pointer-events: none` на элементы формы, т.к. изменение состояния ховера у них сохраняется и это может ввести пользователя в заблуждение, что форма работает но сломалась.

Также все интерактивные элементы формы можно получить через поле `elements` формы.
```suggestion
changeDisabledAttribute(adFormNode.elements, true);
changeDisabledAttribute(mapFiltersFormNode.elements, true);
```